### ISSUE TITLE
[flutter_tools] Provide an option for where to place gen_l10n yaml options

### ISSUE THREAD

**Issue by shihaohong** [flutter/flutter#55795]
Once https://github.com/flutter/flutter/pull/55241 is merged, it could be nice to provide options/alternatives for where the `gen_l10n` tool's yaml configuration can be defined. It currently defaults to `l10n.yaml` in the Flutter project's root directory.

**Comment by christopherfujino** [comment]
Closing as won't fix, @shihaohong feel free to re-open if you plan to implement this :)
